src/main/java/app/model/agents/Evasion/EvasionAgent.java

```diff
@@ -0,0 +1,107 @@
+package app.model.agents.Evasion;
+
+import app.controller.graphicsEngine.Ray;
+import app.controller.linAlg.Vector;
+import app.model.Move;
+import app.model.Type;
+import app.model.agents.Agent;
+import app.model.agents.AgentImp;
+import app.model.agents.WallFollow.WallFollowAgent;
+
+public class EvasionAgent extends AgentImp
+{
+    private EvasionStrategy strategy;
+    private boolean guardInView = true;
+
+    public EvasionAgent(Vector position, Vector direction, double radius, Type type, EvasionStrategy strategy)
+    {
+        super(position, direction, radius, type);
+        this.strategy = strategy;
+    }
+
+    public EvasionAgent(Vector position, Vector direction, double radius, Type type){
+        this(position,direction, radius, type, EvasionStrategy.RANDOM);
+    }
+
+    @Override
+    public Move move()
+    {
+        switch(strategy)
+        {
+            case AWAY -> {
+                return moveAway();
+            }
+            case RANDOMDIRECTED -> {
+                return  moveRandomDirected();
+            }
+            default -> {
+                return moveRandom();
+            }
+        }
+    }
+
+    private Move moveRandom(){
+        int theta = (int) (Math.random() * 360);
+        Vector randomDirection = new Vector(0, 1).rotate(theta);
+
+        return new Move(direction, randomDirection.scale(maxSprint));
+    }
+
+    private Move moveRandomDirected(){
+        // randomness should be a number between 0 and 1
+        Double randomness = 0.3;
```

Comment:
This might be worth making a field variable so that it's easy to find and change?